### PR TITLE
API for libobjects with locality + use it to support localities for some commands

### PR DESCRIPTION
**Pull request by SkySkimmer**
Overlays:
- https://github.com/coq-community/reduction-effects/pull/22

### CODE REVIEW

**Review by gares — APPROVED** [review]

**Comment by Alizter** [comment]
Will we have support for import filters with these commands?

**Comment by SkySkimmer** [comment]
I didn't change any import filter behaviour

File: doc/changelog/08-vernac-commands-and-options/19362-obj-with-locality.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Added:**
+  :cmd:`Register`, :cmd:`Register Scheme` and :cmd:`Add Zify`
+  support attributes :attr:`local`, :attr:`export` and :attr:`global`
```
Comment (jfehrle) [inline doc/changelog/08-vernac-commands-and-options/19362-obj-with-locality.rst:3]:
```suggestion
  now support attributes :attr:`local`, :attr:`export` and :attr:`global`
```

**Review by ppedrot — APPROVED** [review]

**Comment by Janno** [comment]
Could this also be used for `Goptions.MakeRefTable`? It's a very useful functor but unfortunately it lacks support for `#[global`.

**Comment by SkySkimmer** [comment]
In a followup PR, this one should be ready.

File: vernac/vernacentries.ml [hunk 2]
```diff
@@ -2103,8 +2105,11 @@ let vernac_register qid r =
       | GlobRef.IndRef ind -> Global.register_inductive ind pind
       | _ -> CErrors.user_err ?loc:qid.loc (Pp.str "Register in kernel: expecting an inductive type.")
     end
-    else Coqlib.register_ref (Libnames.string_of_qualid n) gr
+    else
+      let local = Attributes.parse hint_locality_default_superglobal atts in
+      Coqlib.register_ref local (Libnames.string_of_qualid n) gr
```
Comment (SkySkimmer) [inline vernac/vernacentries.ml:2110]: this preserves bug https://github.com/coq/coq/issues/19389

**Comment by SkySkimmer** [comment]
https://github.com/coq/coq/pull/19390

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**